### PR TITLE
verify staleness of vendored cluster-api-actuator-pkg

### PR DESCRIPTION
**Pull request by paulfantom**
Simple way to verify if we are vendoring latest actuator-pkg. This should be executed as a separate optional CI job.

/assign	@enxebre @michaelgugino

### CODE REVIEW

**Comment by enxebre** [comment]
are we running `check` on CI? we woudn't want this to block

**Comment by paulfantom** [comment]
We are not. Idea is to implement it in Makefile and later create a non-blocking CI job which will just notify us if we are running tests with latest revision of actuator-pkg or not.

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,11 @@ else
 endif
 
 .PHONY: check
-check: lint fmt vet verify-codegen test ## Run code validations
+check: lint fmt vet verify-codegen check-pkg test ## Run code validations
+
+.PHONY: check-pkg
+check-pkg:
+	dep check | grep -q cluster-api-actuator-pkg && exit 1 || exit 0
```
Comment (ingvagabund) [inline Makefile]: Given we run only `make test-e2e` and `make k8s-e2e` in the CI, it's still useful to have this run by hand.

**Comment by ingvagabund** [comment]
```
$ make check-pkg
dep check | grep -q cluster-api-actuator-pkg && exit 1 || exit 0
dep: check: no such command
Dep is a tool for managing dependencies for Go projects

Usage: "dep [command]"

Commands:

  init     Set up a new Go project, or migrate an existing one
  status   Report the status of the project's dependencies
  ensure   Ensure a dependency is safely vendored in the project
  prune    Pruning is now performed automatically by dep ensure.
  version  Show the dep version information

Examples:
  dep init                               set up a new project
  dep ensure                             install the project's dependencies
  dep ensure -update                     update the locked versions of all dependencies
  dep ensure -add github.com/pkg/errors  add a dependency to the project

Use "dep help [command]" for more information about a command.
```

Am I missing something?

**Comment by paulfantom** [comment]
Looks like your `dep` is old:

```
$ dep
Dep is a tool for managing dependencies for Go projects

Usage: "dep [command]"

Commands:

  init     Set up a new Go project, or migrate an existing one
  status   Report the status of the project's dependencies
  ensure   Ensure a dependency is safely vendored in the project
  version  Show the dep version information
  check    Check if imports, Gopkg.toml, and Gopkg.lock are in sync

Examples:
  dep init                               set up a new project
  dep ensure                             install the project's dependencies
  dep ensure -update                     update the locked versions of all dependencies
  dep ensure -add github.com/pkg/errors  add a dependency to the project

Use "dep help [command]" for more information about a command.
```

File: Makefile (hunk 1 above)
Comment (paulfantom) [reply Makefile]: This will also be run in CI, but as a separate optional job.

**Comment by ingvagabund** [comment]
Running over mao:

```sh
$ make check-pkg
dep check | grep -q cluster-api-actuator-pkg && exit 1 || exit 0
Makefile:28: recipe for target 'check-pkg' failed
make: *** [check-pkg] Error 1
```

**Comment by paulfantom** [comment]
That's correct output. This repo is not using latest revision of actuator-pkg tests suite.

**Pull request merged**